File: codeup/100000575/I.cpp
#include <iostream>
using namespace std;

char get_max_win_key(int *list) {
  int max_win_id = 0;
  char max_win_key[3] = {'B', 'C', 'J'};
  for (int i = 1; i < 3; i++) {
    if (list[i] > list[max_win_id]) max_win_id = i;
  }
  return max_win_key[max_win_id];
}
int main() {
  int N, A_list[3] = {0}, A_win_list[3] = {0}, B_win_list[3] = {0};
  char a, b;
  scanf("%d", &N);
  getchar();
  while (N--) {
    scanf("%c %c", &a, &b);
    getchar();
    if (a == 'B') {
      if (b == 'B')
        A_list[1] += 1;
      else if (b == 'C')
        A_list[0] += 1, A_win_list[0] += 1;
      else if (b == 'J')
        A_list[2] += 1, B_win_list[2] += 1;
    } else if (a == 'C') {
      if (b == 'B')
        A_list[2] += 1, B_win_list[0] += 1;
      else if (b == 'C')
        A_list[1] += 1;
      else if (b == 'J')
        A_list[0] += 1, A_win_list[1] += 1;
    } else if (a == 'J') {
      if (b == 'B')
        A_list[0] += 1, A_win_list[2] += 1;
      else if (b == 'C')
        A_list[2] += 1, B_win_list[1] += 1;
      else if (b == 'J')
        A_list[1] += 1;
    }
  }

  printf("%d %d %d\n", A_list[0], A_list[1], A_list[2]);
  printf("%d %d %d\n", A_list[2], A_list[1], A_list[0]);
  printf("%c %c\n", get_max_win_key(A_win_list), get_max_win_key(B_win_list));

  return 0;
}

File: codeup/100000575/I.py
#!/usr/bin/python3
A_list = [0, 0, 0]
A_win_dict = {'B': 0, 'C': 0, 'J': 0}
B_win_dict = {'B': 0, 'C': 0, 'J': 0}
N = int(input())

while N > 0:
    N -= 1
    a, b = input().split()
    if a == 'B':
        if b == 'B':
            A_list[1] += 1
        elif b == 'C':
            A_list[0] += 1
            A_win_dict[a] += 1
        elif b == 'J':
            A_list[2] += 1
            B_win_dict[b] += 1
    elif a == 'C':
        if b == 'B':
            A_list[2] += 1
            B_win_dict[b] += 1
        elif b == 'C':
            A_list[1] += 1
        elif b == 'J':
            A_list[0] += 1
            A_win_dict[a] += 1
    elif a == 'J':
        if b == 'B':
            A_list[0] += 1
            A_win_dict[a] += 1
        elif b == 'C':
            A_list[2] += 1
            B_win_dict[b] += 1
        elif b == 'J':
            A_list[1] += 1

print(A_list[0], A_list[1], A_list[2])
print(A_list[2], A_list[1], A_list[0])


def get_max_win_key(win_dict):
    max_win_key = 'B'
    for key in win_dict:
        if win_dict[key] > win_dict[max_win_key]:
            max_win_key = key

    return max_win_key


print(get_max_win_key(A_win_dict), get_max_win_key(B_win_dict))
